benches/throughput.rs

```diff
@@ -50,32 +51,95 @@ struct TestEnv {
     hbone: TcpStream,
 }
 
-/// initialize_environment sets up a benchmarking environment. This works around issues in async setup with criterion.
+/// initialize_environment sets up a bench marking environment. This works around issues in sync setup with criterion.
 /// Since tests are only sending data on existing connections, we setup a connection for each test type in the setup phase.
 /// Tests consume the
-fn initialize_environment(mode: Mode) -> (Arc<Mutex<TestEnv>>, Runtime) {
+
+const N_RULES: usize = 10;
+const N_POLICIES: usize = 10_000;
+const DUMMY_NETWORK: &str = "198.51.100.0/24";
+
+fn create_test_policies() -> Vec<Authorization> {
+    let mut policies: Vec<Authorization> = vec![];
+    let mut rules = vec![];
+    for _ in 0..N_RULES {
+        rules.push(vec![vec![RbacMatch {
+            namespaces: vec![
+                StringMatch::Prefix("random-prefix-2b123".to_string()),
+                StringMatch::Suffix("random-postix-2b723".to_string()),
+                StringMatch::Exact("random-exac-2bc13".to_string()),
+            ],
+            not_namespaces: vec![],
+            principals: vec![
+                StringMatch::Prefix("random-prefix-2b123".to_string()),
+                StringMatch::Suffix("random-postix-2b723".to_string()),
+                StringMatch::Exact("random-exac-2bc13".to_string()),
+            ],
+            not_principals: vec![],
+            source_ips: vec![DUMMY_NETWORK.parse().unwrap()],
+            not_source_ips: vec![],
+            destination_ips: vec![DUMMY_NETWORK.parse().unwrap()],
+            not_destination_ips: vec![],
+            destination_ports: vec![0],
+            not_destination_ports: vec![],
+        }]]);
+    }
+
+    for i in 0..N_POLICIES {
+        policies.push(Authorization {
+            name: format!("policy {i}"),
+            action: ztunnel::rbac::RbacAction::Deny,
+            scope: ztunnel::rbac::RbacScope::Global,
+            namespace: "default".to_string(),
+            rules: rules.clone(),
+        });
+    }
+
+    return policies;
+}
+
+fn initialize_environment(
+    mode: Mode,
+    policies: Vec<Authorization>,
```

Comment:
What would `None` represent then? No policies? I would find that confusing because there are then two ways of representing no policies `None` and `vec![]`.